esmvaltool/diag_scripts/pv_capacityfactor/pv_capacity_factor.R

```diff
@@ -0,0 +1,293 @@
+# #############################################################################
+# diagnostic.R
+# Authors:       Irene Cionni (ENEA, Italy)
+# #############################################################################
+# Description
+# This script modifies the wind capacity factor script wrote in bsc_magic
+#
+# Required
+# season: String to include shortcut for season in plot title
+#         and name (e.g. "djf"). It will be converted to upper case.
+#         This season should be the one set in the preprocessor, since it is
+#         only used as a string and does not affect the data in the diagnostic.
+#         In the default recipe this is solved through a node anchor.
+#
+# Optional
+# maxval_colorbar: Optional upper limit for the colorbar.
+#
+# Caveats
+#
+# Modification history
+#    20210401-cionni_irene: written for v2.0
+#    20210401-weigel_katja: changed to allow multiple models
+#    20210621-weigel_katja: formattiong updates
+#
+# ############################################################################
+library(abind)
+library(climdex.pcic)
+library(ggplot2)
+library(multiApply) # nolint
+library(ncdf4)
+library(RColorBrewer) # nolint
+library(s2dverification)
+library(yaml)
+
+# Parsing input file paths and creating output dirs
+args <- commandArgs(trailingOnly = TRUE)
+params <- read_yaml(args[1])
+initial.options <- commandArgs(trailingOnly = FALSE)
+file_arg_name <- "--file="
+script_name <- sub(
+  file_arg_name, "",
+  initial.options[grep(file_arg_name, initial.options)]
+)
+script_dirname <- dirname(script_name)
+
+source(file.path(script_dirname, "PV_CF.R"))
+plot_dir <- params$plot_dir
+run_dir <- params$run_dir
+work_dir <- params$work_dir
+
+# setup provenance file and list
+provenance_file <- paste0(run_dir, "/", "diagnostic_provenance.yml")
+provenance <- list()
+
+## Create working dirs if they do not exist
+dir.create(plot_dir, recursive = TRUE)
+dir.create(run_dir, recursive = TRUE)
+dir.create(work_dir, recursive = TRUE)
+input_files_per_var <- yaml::read_yaml(params$input_files[1])
+var_names <- names(input_files_per_var)
+model_names <- lapply(input_files_per_var, function(x) x$dataset)
+model_names <- unname(model_names)
+var0 <- lapply(input_files_per_var, function(x) x$short_name)
+fullpath_filenames <- names(var0)
+input_files_per_var1 <- yaml::read_yaml(params$input_files[2])
+var_names1 <- names(input_files_per_var1)
+var1 <- lapply(input_files_per_var1, function(x) x$short_name)
+fullpath_filenames1 <- names(var1)
+var0 <- unname(var0)[1]
+var1 <- unname(var1)[1]
+start_years <- lapply(input_files_per_var, function(x) x$start_year)
+start_years <- unname(start_years)
+end_years <- lapply(input_files_per_var, function(x) x$end_year)
+end_years <- unname(end_years)
+seasons <- toupper(params$season)
+#power_curves <- params$power_curves
+
+var0 <- unlist(var0)
+for (i in seq(1, length(model_names), 1)) {
```

Comment:
I think it is seq(from, to, by= ), so 1 is both the starting point and the increment. But it would not be necessary to give it as increment, because 1 is also the default increment, so I removed it.